src/codegen/compile.ml

```diff
@@ -9411,7 +9411,12 @@ module FuncDec = struct
     Func.define_built_in env name ["env", I32Type] [] (fun env ->
         G.i (LocalGet (nr 0l)) ^^
         ContinuationTable.recall env ^^
-        G.i Drop);
+        Arr.load_field env 2l ^^ (* get the cleanup closure *)
```

Comment:
This is the meat.

Reply:
I guess we are now open to leaking entries in the closure table if the user-defined cleanup traps.
I don't see how one could avoid this apart from maybe delaying the  user-defined cleanup to run on entry to the next message. Probably too complicated to consider. Let's just live with the possible leak.

Reply:
Whatever way to look at it, the last cleanup measure — if failing by trap — will be vulnerable. Leaking a harmless entry (which will never be activated by the replica), and also missing the actual release of lock is the penalty for trapping in this place. But the cross-section of this code is deliberately a small one as it will be hard to pile code into the `finally`. I hope.